diff --git a/crates/atuin-client/src/database.rs b/crates/atuin-client/src/database.rs
index ee8b970..6547bea 100644
--- a/crates/atuin-client/src/database.rs
+++ b/crates/atuin-client/src/database.rs
@@ -540,6 +540,16 @@ impl Database for Sqlite {
                     }
                     &mut query
                 }
+                FilterMode::SessionHost => {
+                    query.and_where_eq("session", quote(&context.session));
+                    if let Some(session_start) = session_start {
+                        query.or_where(format!(
+                            "hostname = {} and timestamp < {session_start}",
+                            quote(&context.hostname)
+                        ));
+                    }
+                    &mut query
+                }
                 FilterMode::Directory => query.and_where_eq("cwd", quote(&context.cwd)),
                 FilterMode::Workspace => query.and_where_like_left("cwd", &git_root),
             };
@@ -655,6 +665,16 @@ impl Database for Sqlite {
                 }
                 &mut sql
             }
+            FilterMode::SessionHost => {
+                sql.and_where_eq("session", quote(&context.session));
+                if let Some(session_start) = session_start {
+                    sql.or_where(format!(
+                        "lower(hostname) = {} and timestamp < {session_start}",
+                        quote(context.hostname.to_lowercase())
+                    ));
+                }
+                &mut sql
+            }
             FilterMode::Directory => sql.and_where_eq("cwd", quote(&context.cwd)),
             FilterMode::Workspace => sql.and_where_like_left("cwd", git_root),
         };
diff --git a/crates/atuin-client/src/settings.rs b/crates/atuin-client/src/settings.rs
index 40a07ec..7f472d0 100644
--- a/crates/atuin-client/src/settings.rs
+++ b/crates/atuin-client/src/settings.rs
@@ -130,6 +130,9 @@ pub enum FilterMode {
 
     #[serde(rename = "session-preload")]
     SessionPreload = 5,
+
+    #[serde(rename = "session-host")]
+    SessionHost = 6,
 }
 
 impl FilterMode {
@@ -141,6 +144,7 @@ impl FilterMode {
             FilterMode::Directory => "DIRECTORY",
             FilterMode::Workspace => "WORKSPACE",
             FilterMode::SessionPreload => "SESSION+",
+            FilterMode::SessionHost => "SESSION*",
         }
     }
 }
@@ -772,6 +776,7 @@ impl Default for Search {
                 FilterMode::Host,
                 FilterMode::Session,
                 FilterMode::SessionPreload,
+                FilterMode::SessionHost,
                 FilterMode::Workspace,
                 FilterMode::Directory,
             ],
@@ -1546,6 +1551,7 @@ impl Settings {
                     "workspace",
                     "directory",
                     "session-preload",
+                    "session-host",
                 ],
             )?
             .set_default("theme.name", "default")?
diff --git a/crates/atuin-daemon/src/client.rs b/crates/atuin-daemon/src/client.rs
index a37fa57..2e7c0d1 100644
--- a/crates/atuin-daemon/src/client.rs
+++ b/crates/atuin-daemon/src/client.rs
@@ -272,6 +272,7 @@ impl From<FilterMode> for RpcFilterMode {
             FilterMode::Directory => RpcFilterMode::Directory,
             FilterMode::Workspace => RpcFilterMode::Workspace,
             FilterMode::SessionPreload => RpcFilterMode::SessionPreload,
+            FilterMode::SessionHost => RpcFilterMode::Session,
         }
     }
 }
