--- a/crates/atuin-client/src/database.rs
+++ b/crates/atuin-client/src/database.rs
@@ -587,6 +587,16 @@
                     }
                     &mut query
                 }
+                FilterMode::SessionHost => {
+                    query.and_where_eq("session", quote(&context.session));
+                    if let Some(session_start) = session_start {
+                        query.or_where(format!(
+                            "{} and timestamp < {session_start}",
+                            origin_sql_filter(&context.cmd_origin)
+                        ));
+                    }
+                    &mut query
+                }
                 FilterMode::Directory => query.and_where_eq("cwd", quote(&context.cwd)),
                 FilterMode::Workspace => query.and_where_like_left("cwd", &git_root),
             };
@@ -705,6 +715,16 @@
                 }
                 &mut sql
             }
+            FilterMode::SessionHost => {
+                sql.and_where_eq("session", quote(&context.session));
+                if let Some(session_start) = session_start {
+                    sql.or_where(format!(
+                        "{} and timestamp < {session_start}",
+                        origin_sql_filter(&context.cmd_origin)
+                    ));
+                }
+                &mut sql
+            }
             FilterMode::Directory => sql.and_where_eq("cwd", quote(&context.cwd)),
             FilterMode::Workspace => sql.and_where_like_left("cwd", git_root),
         };
--- a/crates/atuin-client/src/settings.rs
+++ b/crates/atuin-client/src/settings.rs
@@ -136,6 +136,9 @@
 
     #[serde(rename = "session-preload")]
     SessionPreload = 5,
+
+    #[serde(rename = "session-host")]
+    SessionHost = 6,
 }
 
 impl FilterMode {
@@ -148,6 +151,7 @@
             Self::Directory => "DIRECTORY",
             Self::Workspace => "WORKSPACE",
             Self::SessionPreload => "SESSION+",
+            Self::SessionHost => "SESSION*",
         }
     }
 }
@@ -755,6 +759,7 @@
                 FilterMode::Host,
                 FilterMode::Session,
                 FilterMode::SessionPreload,
+                FilterMode::SessionHost,
                 FilterMode::Workspace,
                 FilterMode::Directory,
             ],
@@ -1554,6 +1559,7 @@
                 "workspace",
                 "directory",
                 "session-preload",
+                "session-host",
             ])?
             .set_default("theme.name", "default")?
             .set_default("theme.debug", None::<bool>)?
--- a/crates/atuin-daemon/src/client.rs
+++ b/crates/atuin-daemon/src/client.rs
@@ -287,6 +287,7 @@
             FilterMode::Directory => Self::Directory,
             FilterMode::Workspace => Self::Workspace,
             FilterMode::SessionPreload => Self::SessionPreload,
+            FilterMode::SessionHost => Self::Session,
         }
     }
 }
